### DISCUSSION TITLE
Migrating legacy project

### DISCUSSION THREAD

**Discussion by realJogicodes** [pocketbase/pocketbase#4329] [Q&A]
Hi

I've got a legacy project that I am considering porting over to pocketbase because pocketbase is awesome. It's so much better than what the legacy project uses which is plain sqlite3 with a python/flask/flask_sqlalchemy backend. 

User data is stored in a table and the pasword is stored as a hash with flask_bcrypt:

`form = FlaskForm()`
`password = form.password.data`
`pw_hash = bcrypt.generate_password_hash(password)`
`user=User(...,pw_hash)`
`db.session.add(user)`
`db.session.commit()`

Is there a way to migrate my user model to pb without asking my legacy users to reset their passwords?

What I though of doing is make a table that has all previous signups via email in it that then checks during login if the user is known and then creates a new pockerbase user with the login password on the first attept. I think that would work but I also think it kinda is a little sketchy - mildly put - and there must be a better way?

**Comment by ganigeorgiev** [comment]
> Is there a way to migrate my user model to pb without asking my legacy users to reset their passwords?

PocketBase also uses bcrypt for the password hash so it should be compatible without requiring users to reset their passwords.

My recommendation would be to try to recreate your existing db structure in PocketBase, then explore the generated SQLite tables and try to adapt your migration script to it (_note that auth collections requires each user to have a random unique `tokenKey` column_).

**Reply by realJogicodes** [reply]
That's great to read. Will try next week. My old user.id are integers, ascending. That's not gonna fly with the pocketbase ID, which I think is what you meant with tokenKey?

So I would have to first duplicate the user table in a way where ID is replaced with the pocketbase way of doing IDs (is that a UUID?)

**Reply by ganigeorgiev** [reply]
> That's not gonna fly with the pocketbase ID, which I think is what you meant with tokenKey?

You can insert the old record id as it is, it doesn't need to match with the 15 chars limit (_or alternatively if you want to match you can pad the existing value_).

The `tokenKey` is a unique random string that is used as part of the tokens secret.

**Reply by realJogicodes** [reply]
Thanks for the clarification
